src/DetailsView/reports/components/outcome-summary-bar.tsx

```diff
@@ -1,33 +1,37 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { kebabCase } from 'lodash';
 import * as React from 'react';
 
 import { NamedSFC } from '../../../common/react/named-sfc';
-import { OutcomeIcon } from './outcome-icon';
-import { allRequirementOutcomeTypes, RequirementOutcomeStats } from './requirement-outcome-type';
+import { outcomeIconMap, outcomeIconMapInverted, OutcomeStats, OutcomeType, outcomeTypeSemantics, OutcomeUnits } from './outcome-type';
 
-type OutcomeUnits = 'percentage' | 'requirements';
-
-const outcomeText = {
-    pass: 'Passed',
-    incomplete: 'Incomplete',
-    fail: 'Failed',
+export type OutcomeSummaryBarProps = {
+    outcomeStats: Partial<OutcomeStats>;
+    allOutcomeTypes: OutcomeType[];
+    units?: OutcomeUnits;
+    inverted?: boolean;
 };
 
-export type OutcomeSummaryBarProps = RequirementOutcomeStats & { units?: OutcomeUnits };
+export const OutcomeSummaryBar = NamedSFC<OutcomeSummaryBarProps>('OutcomeSummaryBar', props => {
+    return (
+        <div className="outcome-summary-bar">
+            {props.allOutcomeTypes.map(outcomeType => {
+                const { units, inverted } = props;
+                const text = outcomeTypeSemantics[outcomeType].pastTense;
```

Comment:
`outcomeTypeSemantics` is a hard dependency here (we're importing it from a different file and using it directly on the component).
I wonder if it make sense to receive this map as a prop instead...

Reply:
Do you want to put it in the deps? I feel this is fine here since it is just a static object. But I'm open for changes.

Reply:
It's an static object we're using as a global... not sure that's a pattern we want to follow. Also, we need to consider that right now, the object has definition for 2 sets of slightly different things, requirements and instances. Breaking this object apart will enable us to update/modified this concepts separately and the I guess one of the first step to do that in the near future is to not have a hard dependency on the object. Up to you.

Reply:
Sounds good. I'll leave this PR as it is since there is already 20+ file changes.